Add cast methods ToUintSlice

nmvalera opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/spf13/cast?pullRequest=236) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/spf13/cast?pullRequest=236) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/spf13/cast?pullRequest=236) it.</sub>

sagikazarmark reviewed (changes requested):
@nmvalera could you please reabase your PR? I added a set of generic functions for slices, could you please use those instead?

Thanks!

sagikazarmark commented:
Overlaps with #179

nmvalera commented:
Hey @sagikazarmark, I have rebased to use the slice utilities.

I also noticed that ToFloat64Slice has been implemented in the meantime, so I removed this part from my PR (I just left some extra test cases that were on my PR, as it does not hurt)

sagikazarmark reviewed (approved):
Thanks!
